log4j-appender/src/main/java/org/apache/kafka/log4jappender/KafkaLog4jAppender.java

```diff
@@ -333,6 +333,7 @@ public void activateOptions() {
         props.put(KEY_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
         props.put(VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
         this.producer = getKafkaProducer(props);
+        LogLog.warn("log4j-appender is deprecated and will be removed in a future release. For migration, please refer to the latest documentation.");
```

Comment:
Should we also annotate the class with `@Deprecated`?

Reply:
Well, although this class will not be instantiated directly, it would be much better.